core/src/main/java/org/pentaho/di/trans/steps/cassandraoutput/CassandraOutput.java

```diff
@@ -373,6 +373,10 @@ protected void doBatch( List<Object[]> batch ) throws Exception {
       int rowsAdded = 0;
       batch = CassandraUtils.fixBatchMismatchedTypes( batch, getInputRowMeta(), m_cassandraMeta );
       DriverCQLRowHandler handler = (DriverCQLRowHandler) m_cqlHandler;
+      String ttl = m_meta.getTTL();
+      if ( !Utils.isEmpty( ttl ) ) {
+        handler.setTtlSec( Integer.parseInt( ttl ) );
```

Comment:
Is it possible for the ttl to be different from an Integer value? (by wrong user input for example). If so, what happens here  if Interger.parseInt throws an exception?

Reply:
i recall testing this at the time, let me check if this validation is already made by our interface

Reply:
This is already validated during the cassandra database connection process. That's why i did not see this before providing my fix. Thank you. New commit added using the same validation logic on this one also.